File: src/main/java/recipeapplication/utility/RecipeTime.java
package recipeapplication.utility;

public class RecipeTime
{
    private static final String DEFAULT_VALUE = "00:00";

    public static String combineCookAndPrepTime(String cookTime, String prepTime)
    {
        // TODO I might want to handle invalid formats

        if (cookTime.isEmpty() && prepTime.isEmpty())
        {
            return DEFAULT_VALUE;
        }

        if (cookTime.isEmpty())
        {
            cookTime = DEFAULT_VALUE;
        }
        else if (prepTime.isEmpty())
        {
            prepTime = DEFAULT_VALUE;
        }

        cookTime = cookTime.replace(":", "");
        prepTime = prepTime.replace(":", "");

        String totalTime = addCookAndPrepTime(cookTime, prepTime);
        totalTime = addColonToTime(totalTime);

        return totalTime;
    }

    private static String zeroPadTime4DP(String string)
    {
        return String.format("%04d", Integer.parseInt(string));
    }

    private static String zeroPadTime2DP(int time)
    {
        return String.format("%02d", time);
    }

    private static String addCookAndPrepTime(String cookTime, String prepTime)
    {
        String totalTime = Integer.toString(Integer.parseInt(cookTime) + Integer.parseInt(prepTime));

        int totalTimeMinutes = getMinuteFromTime(totalTime);
        int totalTimeHours = getHourFromTime(totalTime);

        if (totalTimeMinutes > 59)
        {
            totalTimeMinutes = totalTimeMinutes - 60;
            totalTimeHours++;
        }

        return zeroPadTime2DP(totalTimeHours) + zeroPadTime2DP(totalTimeMinutes);
    }

    private static String addColonToTime(String time)
    {
        return time.substring(0, 2) + ":" + time.substring(2);
    }

    private static int getHourFromTime(String time)
    {
        String hoursAsString = zeroPadTime4DP(time).substring(0, 2);

        if (!hoursAsString.equals(""))
        {
            return Integer.parseInt(hoursAsString);
        }

        return 0;
    }

    private static int getMinuteFromTime(String time)
    {
        String zeroPaddedTime = zeroPadTime4DP(time);

        return Integer.parseInt(zeroPaddedTime.substring(zeroPaddedTime.length() -2));
    }
}

File: src/main/kotlin/thecookingpot/utility/RecipeTime.kt
package thecookingpot.utility

private const val DEFAULT_VALUE = "00:00"

fun combineCookAndPrepTime(cookTime: String, prepTime: String): String {
    if (cookTime.isEmpty() && prepTime.isEmpty()) {
        return DEFAULT_VALUE
    }
    if (cookTime.isEmpty()) {
        return prepTime
    } else if (prepTime.isEmpty()) {
        return cookTime
    }

    val totalTime = addCookAndPrepTime(
            cookTime.replace(":", ""),
            prepTime.replace(":", "")
    )

    return addColonToTime(totalTime)
}

private fun zeroPadTime4DP(string: String): String {
    return "%04d".format(string.toInt())
}

private fun zeroPadTime2DP(time: Int): String {
    return "%02d".format(time)
}

private fun addCookAndPrepTime(cookTime: String, prepTime: String): String {
    val totalTime = (cookTime.toInt() + prepTime.toInt()).toString()
    var totalTimeMinutes = getMinuteFromTime(totalTime)
    var totalTimeHours = getHourFromTime(totalTime)
    if (totalTimeMinutes > 59) {
        totalTimeMinutes -= 60
        totalTimeHours++
    }
    return zeroPadTime2DP(totalTimeHours) + zeroPadTime2DP(totalTimeMinutes)
}

private fun addColonToTime(time: String): String {
    return time.substring(0, 2) + ":" + time.substring(2)
}

private fun getHourFromTime(time: String): Int {
    val hoursAsString = zeroPadTime4DP(time).substring(0, 2)
    return hoursAsString.toInt()
}

private fun getMinuteFromTime(time: String): Int {
    val zeroPaddedTime = zeroPadTime4DP(time)
    return zeroPaddedTime.substring(zeroPaddedTime.length - 2).toInt()
}
